components/Filter/Index.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { theme } from "@/styles";
+
+export default function Filter({ onSubmit, showFilterWindow }) {
+  const [author, setAuthor] = useState("");
+  const [category, setCategory] = useState("");
+
+  const handleAuthorChange = (event) => {
+    const value = event.target.value;
+    setAuthor(value);
+  };
+
+  const handleCategoryChange = (event) => {
+    const value = event.target.value;
+    setCategory(value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    onSubmit({ author, category });
+  };
+
+  return (
+    <StyledFilterForm show={showFilterWindow} onSubmit={handleSubmit}>
+      <StyledFilterBox>
+        <h2>Filter</h2>
+        <StyledCloseButton onClick={() => (window.location.href = "/")}>
+          x
+        </StyledCloseButton>
+        <StyledUl>
+          <StyledFilterLabel>Author: </StyledFilterLabel>
+          <StyledFilterInput
+            type="search"
+            placeholder="Search..."
+            value={author}
+            onChange={handleAuthorChange}
+          />
+          <StyledFilterLabel>Category:</StyledFilterLabel>
+          <select
+            id="category"
+            onChange={handleCategoryChange}
+            value={category}
+          >
+            <option value="">all categories</option>
+            <option value="Sports">Sports</option>
+            <option value="Culture">Culture</option>
+            <option value="Food">Food</option>
+            <option value="Outdoor">Outdoor</option>
+          </select>
+          <StyledFilterLabel>Weekday:</StyledFilterLabel>
+          <div>-- coming soon --</div>
+          <StyledFilterLabel>Time:</StyledFilterLabel>
+          <div>-- coming soon --</div>
+          <StyledFilterLabel>Day:</StyledFilterLabel>
+          <div>-- coming soon --</div>
+        </StyledUl>
+        <StyledFilterButton type="submit">ok</StyledFilterButton>
+      </StyledFilterBox>
+    </StyledFilterForm>
+  );
+}
+
+const StyledFilterForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  margin: ${theme.spacing.small} auto;
+  max-width: calc(${theme.box.width} + 10px);
+  padding-left: 5px;
+  overflow: hidden;
+
+  transition: max-height 0.8s cubic-bezier(0.645, 0.045, 0.355, 1);
+
+  max-height: ${(props) => (props.show ? "500px" : "0")};
```

Comment:
you should prefix props passed to a styled component with a "$". Otherwise they will get passed to the HTML element as an attribute which will cause an error in the console, since "show" is not a native HTML attribute:

```suggestion
  max-height: ${(props) => (props.$show ? "500px" : "0")};
```

Reply:
I tried that but it doesn't work. This way the filter won't slide out.